### ISSUE TITLE
Clipping ratio and response length blow-up on GSM8k with default Trinity config

### ISSUE THREAD

**Issue by vadimkantorov** [modelscope/Trinity-RFT#176]
Hi!

We're getting response length and clip ratio blow-up on GSM8k which leads to halting of any learning.

We've tried the default Trinity GRPO config (except we put the number of epochs to a large value)

By chance, do you have any traces of longer wandb runs published for for the baseline configs GSM8k?

Would you have any suggestions on preventing such blow-up?

Thank you :)

<img width="612" height="476" alt="Image" src="https://github.com/user-attachments/assets/b6c91fb6-13e9-4baa-846e-747da227d116" />

**Comment by pan-x-c** [comment]
Set the learning rate to `1e-6` may help. This example is a bit stale. We will provide a version corresponding to the current code next week.

**Comment by yanxi-chen** [comment]
Hi! 

Could you also clarify which "number of epoch" are you referring to when you say "we put the number of epochs to a large value"? 

Increasing `ppo_epochs` in `train_gsm8k.yaml` can be problematic (at least risky) and not recommended in general. Setting this to a large value means the same batch of experiences will be reused for multiple times consecutively (you may see this in the line `for epoch in range(self.config.ppo_epochs)` within `dp_actor.py`), which can easily lead to overfitting and instability. 

If you intend to run RL on GSM8k for multiple epochs, you should increase `buffer: total_epochs` in `gsm8k.yaml` instead.

**Comment by nkkarpov** [comment]
Thank you for the fast response!

Thank you for the clarification, @yanxi-chen, we have used ```buffer: total_epochs```. 
@pan-x-c do you have any quick explanation why this example is stale?  Did you have some major fixes that could effect this example?

**Comment by yanxi-chen** [comment]
Thanks for the clarification @nkkarpov.

We will try and see if we can reproduce this blow-up of response length by the end of next Monday, as the Trinity team members are currently out of office for weekend :)

Meanwhile, it would be helpful if you can present further details about your experiments, e.g., full configs of `gsm8k.yaml` and `train_gsm8k.yaml`, wandb metrics for min/max/mean response length, and any metric that seems suspicious.

**Comment by yanxi-chen** [comment]
We have just re-run the GSM8k experiment, using the most updated codebase and the exact configs in `examples/grpo_gsm8k`. The results look normal though, with increasing mean reward and stable response length. Perhaps more details about your experiment setup and results could help us better pinpoint the issue 🤔

<img width="1044" height="290" alt="Image" src="https://github.com/user-attachments/assets/0c1a97da-6d25-4552-954c-94bc05102f7c" />

<img width="1058" height="540" alt="Image" src="https://github.com/user-attachments/assets/c1be6d5b-38ee-49ef-af28-ea0fbf4bd474" />

**Comment by vadimkantorov** [comment]
> We have just re-run the GSM8k experiment

Thank you! I'll check again, but I think our failed run also followed the same vanilla config. And we had to decrease the learning rate to avoid clipping

Could you please publish the link to the wandb run?

**Comment by yanxi-chen** [comment]
> Could you please publish the link to the wandb run?

Due to the company's policy regarding data security, we are not able to publish our wandb link at the moment. We will see what we can do about it.

You may have interest in the new "benchmark" feature of Trinity-RFT, please check out [this folder](https://github.com/modelscope/Trinity-RFT/tree/main/benchmark)!

If you have used the same vanilla config in your experiment, it seems that the only variable in that config is the model path. Could you clarify which initial model was used? In our experiment we used Qwen2.5-1.5B-Instruct.

**Comment by vadimkantorov** [comment]
I think we ran it with Qwen3-1.5B, cc @nkkarpov

**Comment by nkkarpov** [comment]
I just tried `Qwen2.5-1.5B-Instruct` and we still having same issues.

<img width="2498" height="651" alt="Image" src="https://github.com/user-attachments/assets/7f061efa-436b-4390-9ecd-b85bf9c22cc8" />

<img width="2498" height="649" alt="Image" src="https://github.com/user-attachments/assets/1d06fa74-3668-44fa-b025-4a17427c64c9" />

about our setup:
``` 
>nvidia-smi | head -4
Fri Aug 29 17:48:45 2025
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.01             Driver Version: 535.183.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
```

We run cluster from the following venv:
```bash
 uv pip list
Using Python 3.12.0 environment at: /home/ubuntu/venvs/ray
Package                            Version
---------------------------------- -----------
aiohappyeyeballs                   2.6.1
aiohttp                            3.12.15
aiohttp-cors                       0.8.1
aiosignal                          1.4.0
annotated-types                    0.7.0
attrs                              25.3.0
cachetools                         5.5.2
certifi                            2025.8.3
charset-normalizer                 3.4.3
click                              8.2.1
colorful                           0.5.7
distlib                            0.4.0
filelock                           3.19.1
frozenlist                         1.7.0
google-api-core                    2.25.1
google-auth                        2.40.3
googleapis-common-protos           1.70.0
grpcio                             1.74.0
idna                               3.10
importlib-metadata                 8.7.0
jsonschema                         4.25.1
jsonschema-specifications          2025.4.1
msgpack                            1.1.1
multidict                          6.6.4
opencensus                         0.11.4
opencensus-context                 0.1.3
opentelemetry-api                  1.36.0
opentelemetry-exporter-prometheus  0.57b0
opentelemetry-proto                1.36.0
opentelemetry-sdk                  1.36.0
opentelemetry-semantic-conventions 0.57b0
packaging                          25.0
platformdirs                       4.4.0
prometheus-client                  0.22.1
propcache                          0.3.2
proto-plus                         1.26.1
protobuf                           6.32.0
py-spy                             0.4.1
pyasn1                             0.6.1
pyasn1-modules                     0.4.2
pydantic                           2.11.7
pydantic-core                      2.33.2
pyyaml                             6.0.2
ray                                2.48.0
referencing                        0.36.2
requests                           2.32.5
rpds-py                            0.27.1
rsa                                4.9.1
six                                1.17.0
smart-open                         7.3.0.post1
typing-extensions                  4.15.0
typing-inspection                  0.4.1
urllib3                            2.5.0
virtualenv                         20.34.0
wrapt                              1.17.3
yarl                               1.20.1
zipp                               3.23.0
```

our branch:
```bash
>git log --oneline  | head
cc42ba1c update config
c5e6dd31 Bug fix for FSDP offload (#233)
fb29ddc2 [example] An email search workflow (#230)
2aef9274 AsymRE (#187)
eebced30 Enhance SFT/DPO reader (#226)
013c2c7f Fix alfworld prompt typo (#222)
4576a55f Fix links in readme (#225)
6c5e18fd Bug fix for experience display in wandb (#224)
d193f201  Enhance Logger (#217)
65b441d5 Add data_process figure (#218)
```

patch file: [0001-update-config.patch](https://github.com/user-attachments/files/22051266/0001-update-config.patch)

we run job from the head node with the following command:
```bash
RAY_ADDRESS='http://127.0.0.1:8265' ray job submit   --runtime-env-json '{"working_dir": ".", "py_executable": "uv run", "excludes": [".git/"], "VLLM_DISABLE_COMPILE_CACHE": "1" }'   --no-wait   -- python -m trinity.cli.launcher run --config examples/grpo_gsm8k/gsm8k.yaml
```

**Comment by pan-x-c** [comment]
Could you try the [Benchmark tool](https://github.com/modelscope/Trinity-RFT/tree/main/benchmark) we provided? There may still be some configuration misalignment here
